flink-python/pyflink/table/tests/test_udtf.py

```diff
@@ -41,44 +41,67 @@ def test_table_function(self):
                              DataTypes.BIGINT()))
 
         t = self.t_env.from_elements([(1, 1, 3), (2, 1, 6), (3, 2, 9)], ['a', 'b', 'c'])
-        t.join_lateral("multi_emit(a, multi_num(b)) as (x, y)") \
+        t = t.join_lateral("multi_emit(a, multi_num(b)) as (x, y)") \
             .left_outer_join_lateral("condition_multi_emit(x, y) as m") \
-            .select("x, y, m") \
-            .insert_into("Results")
-        self.t_env.execute("test")
-        actual = source_sink_utils.results()
+            .select("x, y, m")
+        actual = self._get_output(t)
         self.assert_equals(actual,
                            ["1,0,null", "1,1,null", "2,0,null", "2,1,null", "3,0,0", "3,0,1",
                             "3,0,2", "3,1,1", "3,1,2", "3,2,2", "3,3,null"])
 
     def test_table_function_with_sql_query(self):
-        table_sink = source_sink_utils.TestAppendSink(
+        self._register_table_sink(
             ['a', 'b', 'c'],
             [DataTypes.BIGINT(), DataTypes.BIGINT(), DataTypes.BIGINT()])
-        self.t_env.register_table_sink("Results", table_sink)
+
         self.t_env.register_function(
             "multi_emit", udtf(MultiEmit(), [DataTypes.BIGINT(), DataTypes.BIGINT()],
                                [DataTypes.BIGINT(), DataTypes.BIGINT()]))
 
         t = self.t_env.from_elements([(1, 1, 3), (2, 1, 6), (3, 2, 9)], ['a', 'b', 'c'])
         self.t_env.register_table("MyTable", t)
-        self.t_env.sql_query(
+        t = self.t_env.sql_query(
             "SELECT a, x, y FROM MyTable LEFT JOIN LATERAL TABLE(multi_emit(a, b)) as T(x, y)"
-            " ON TRUE") \
-            .insert_into("Results")
-        self.t_env.execute("test")
-        actual = source_sink_utils.results()
+            " ON TRUE")
+        actual = self._get_output(t)
         self.assert_equals(actual, ["1,1,0", "2,2,0", "3,3,0", "3,3,1"])
 
+    def _register_table_sink(self, field_names: list, field_types: list):
+        pass
+
+    def _get_output(self, t):
+        pass
+
 
 class PyFlinkStreamUserDefinedTableFunctionTests(UserDefinedTableFunctionTests,
                                                  PyFlinkStreamTableTestCase):
-    pass
+
+    def _register_table_sink(self, field_names: list, field_types: list):
+        table_sink = source_sink_utils.TestAppendSink(field_names, field_types)
+        self.t_env.register_table_sink("Results", table_sink)
+
+    def _get_output(self, t):
+        t.insert_into("Results")
+        self.t_env.execute("test")
+        return source_sink_utils.results()
```

Comment:
How about making this two as the default implementation in the base class and only override them in the `PyFlinkBatchUserDefinedTableFunctionTests`?